### PR TITLE
Simple Mobs Do Be Frozen Now: Rewrites the datum AI guide, Converts Butterflies to Basic, Refactors Food Subtree, Renames basic_controller path to basic

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

- Rewrites the AI datum guide. The old guide was so bad holy shit sorry lads. The new one is quick, to the point, hits the core concepts, and doesn't do anything stupid. It's currently a rough draft, so it may change before merge.
- Converts Butterflies to Basic. This was just a small side action I did while writing the AI guide since I wanted to use them as a barebones implementation
- Refactors Food Subtree. This also helped with the AI guide, better combination of subtrees that have the same goal.
- Renames basic_controller path to basic. This was just bad naming on the type left over from early days of datum AI.

## Why It's Good For The Game

This should be more good groundwork for converting simple to basic and further developing basic AI.

## Changelog

:cl:
refactor: refactored butterflies, you might notice some minor differences...?
/:cl:

### CODE REVIEW

File: code/datums/ai/basic_mobs/basic_subtrees/find_and_eat_food.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/// similar to finding a target but looks for food types in the wanted list
+/datum/ai_planning_subtree/find_and_eat_food
+
+/datum/ai_planning_subtree/find_and_eat_food/SelectBehaviors(datum/ai_controller/controller, delta_time)
+	var/datum/weakref/weak_target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+	var/atom/target = weak_target?.resolve()
+	var/list/wanted = controller.blackboard[BB_BASIC_FOODS]
+	if(!target || QDELETED(target))
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/datums/ai/basic_mobs/basic_subtrees/find_and_eat_food.dm:8]: this should definitely be spread around basic way more then nice

File: .github/guides/ai/LEARN_AI.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		controller.queue_behavior(/datum/ai_behavior/find_nearby, BB_MONKEY_CURRENT_PRESS_TARGET)
+		return
+
+	if(prob(50))
+		controller.queue_behavior(/datum/ai_behavior/use_on_object, BB_MONKEY_CURRENT_PRESS_TARGET)
+		return SUBTREE_RETURN_FINISH_PLANNING
+```
+As you can see we're putting the search behavior... on a behavior! The monkey will plan from other subtrees while it finds nearby, but now we have a BB any subtree can look at, any behavior can modify or remove.
+
+## Advanced Concepts
```
Comment (Mothblocks) [inline .github/guides/ai/LEARN_AI.md:174]: armhulen is so confident in his system that he doesnt think ANY concepts are advanced and did not write anything

Comment (tralezab) [reply .github/guides/ai/LEARN_AI.md:174]: ROUGH DRAFT I'M THINKIN

Comment (tralezab) [reply .github/guides/ai/LEARN_AI.md:174]: might put signal hooking into it

File: _maps/map_files/KiloStation/KiloStation.dmm [hunk 3]
```diff
@@ -51619,7 +51619,7 @@
 /obj/structure/flora/bush/lavendergrass/style_random,
 /obj/structure/flora/bush/fullgrass/style_random,
 /obj/structure/flora/bush/grassy/style_random,
-/mob/living/simple_animal/butterfly,
+/mob/living/basic/butterfly,
```
Comment (Mothblocks) [inline _maps/map_files/KiloStation/KiloStation.dmm:51622]: make sure to include one of those path rename scripts

Comment (Mothblocks) [reply _maps/map_files/KiloStation/KiloStation.dmm:51622]: note to self: we should lint for this at some point

Comment (san7890) [reply _maps/map_files/KiloStation/KiloStation.dmm:51622]: Look for a change in paths, then flag with a "Needs UpdatePaths" label if map edits are also made?

Comment (Mothblocks) [reply _maps/map_files/KiloStation/KiloStation.dmm:51622]: Yes something like that

Comment (Mothblocks) [reply _maps/map_files/KiloStation/KiloStation.dmm:51622]: theoretically we can also unit test to make sure that the path update tools when run over your maps dont cause any changes. dunno how often that regresses

Comment (Iamgoofball) [reply _maps/map_files/KiloStation/KiloStation.dmm:51622]: we lint for this it's called the map doesn't compile 😎

**Comment by Mothblocks** [comment]
Armhulen has already given up on basic mobs and is starting /mob/living/basic2

**Comment by optimumtact** [comment]
incoherent screaming

**Comment by tralezab** [comment]
> Armhulen has already given up on basic mobs and is starting /mob/living/basic2

i've started on shrimplemobs

**Comment by Iamgoofball** [comment]
stalebot saving us from the freeze

**Comment by Mothblocks** [comment]
the freeze already happened in my pr

**Comment by Mothblocks** [comment]
but yes reopen when the conflicts and such are fixed, we're stacked on prs

**Comment by Mothblocks** [comment]
prefreably split the docs into its own pr so its easier for me to review

**Pull request closed without merging**